File: frontend/src/types/surface.ts
export type SurfaceTypeValue = 'WALL' | 'CEILING' | 'FLOOR' | 'OTHER';

export interface SurfaceType {
  id: string;
  room_id: string;
  name: string;
  surface_type: SurfaceTypeValue;
  description: string | null;
  position?: number | null;
  width?: string | number | null;
  height?: string | number | null;
  gross_area?: string | number | null;
  deduction_area?: string | number | null;
  net_area?: string | number | null;
  is_archived: boolean;
  created_at: string;
  updated_at: string;
}

export interface SurfaceListResponse {
  items: SurfaceType[];
  total: number;
}

export interface SurfaceCreatePayload {
  name: string;
  surface_type: SurfaceTypeValue;
  description?: string | null;
  position?: number | null;
  width?: number | string | null;
  height?: number | string | null;
}

export interface SurfaceUpdatePayload {
  name?: string;
  surface_type?: SurfaceTypeValue;
  description?: string | null;
  position?: number | null;
  width?: number | string | null;
  height?: number | string | null;
}

File: backend/app/models/surface.py
import enum
import uuid
from datetime import datetime, timezone
from decimal import Decimal

from sqlalchemy import Boolean, DateTime, Enum, ForeignKey, Index, Integer, Numeric, String, Uuid
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base


class SurfaceType(str, enum.Enum):
    WALL = "WALL"
    CEILING = "CEILING"
    FLOOR = "FLOOR"
    OTHER = "OTHER"


class Surface(Base):
    __tablename__ = "surfaces"

    id: Mapped[uuid.UUID] = mapped_column(
        Uuid,
        primary_key=True,
        default=uuid.uuid4,
    )
    room_id: Mapped[uuid.UUID] = mapped_column(
        Uuid,
        ForeignKey("rooms.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )
    name: Mapped[str] = mapped_column(String(255), nullable=False)
    surface_type: Mapped[SurfaceType] = mapped_column(
        Enum(SurfaceType, name="surfacetype"),
        nullable=False,
    )
    description: Mapped[str | None] = mapped_column(String(4096), nullable=True)
    position: Mapped[int | None] = mapped_column(
        Integer,
        nullable=True,
        comment="Sequential order among surfaces of a room (null for legacy surfaces)",
    )
    width: Mapped[Decimal | None] = mapped_column(
        Numeric(10, 3),
        nullable=True,
    )
    height: Mapped[Decimal | None] = mapped_column(
        Numeric(10, 3),
        nullable=True,
    )
    is_archived: Mapped[bool] = mapped_column(Boolean, nullable=False, default=False)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        default=lambda: datetime.now(timezone.utc),
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        default=lambda: datetime.now(timezone.utc),
        onupdate=lambda: datetime.now(timezone.utc),
    )

    __table_args__ = (
        Index("ix_surfaces_room_archived", "room_id", "is_archived"),
    )
